port/zephyr/golioth_fw_zephyr.c

```diff
@@ -240,9 +261,71 @@ enum golioth_status fw_update_post_download(void)
     return GOLIOTH_OK;
 }
 
-enum golioth_status fw_update_validate(void)
+/*
+ * @note This is similar to Zephyr's flash_img_check() but uses the Golioth
+ * port's sha256 API.
+ */
+enum golioth_status fw_update_validate(const uint8_t *hash, const size_t size)
 {
-    return GOLIOTH_OK;
+    enum golioth_status status = GOLIOTH_OK;
+    int err;
+    int buf_size;
+    int pos;
+    uint8_t calc_sha256[GOLIOTH_OTA_COMPONENT_BIN_HASH_LEN];
+
+    err = flash_area_open(UPLOAD_FLASH_AREA_ID,
+                          (const struct flash_area **) &(_flash_img_context.flash_area));
+    if (err)
+    {
+        return GOLIOTH_ERR_IO;
+    };
+
+    golioth_sys_sha256_t hash_ctx = golioth_sys_sha256_create();
+
+    buf_size = sizeof(_flash_img_context.buf);
```

Comment:
Note for other reviewers: This buf size defaults to `CONFIG_IMG_BLOCK_BUF_SIZE` which defaults to 512.